### PR TITLE
ec2_vpc_nat_gateway: increase integration tests coverage

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY
Increase integration test coverage for `ec2_vpc_nat_gateway` and `ec2_vpc_nat_gateway_info`

##### COMPONENT NAME
ec2_vpc_nat_gateway
ec2_vpc_nat_gateway_info

##### ISSUE TYPE

- Test Pull Request

### CODE REVIEW

File: tests/integration/targets/ec2_vpc_nat_gateway/tasks/main.yml [hunk 1]
```diff
@@ -145,6 +145,35 @@
...
+      filters:
+        subnet-id: "{{ subnet_id }}"
+        state: ['available']
+    register: avalaible_ngws
+
+  - name: Assert success
+    assert:
+      that:
+          - avalaible_ngws is successful
+          - avalaible_ngws.result | length == 1
```
Comment (tremble) [inline tests/integration/targets/ec2_vpc_nat_gateway/tasks/main.yml:161]: It'd be good to assert the shape of results too:

```suggestion
    assert:
      that:
          - avalaible_ngws is successful
          - avalaible_ngws.result | length == 1
          - '"create_time" in first_ngw'
          - '"nat_gateway_addresses" in first_ngw'
          - '"nat_gateway_id" in first_ngw'
          - first_ngw.nat_gateway_id == create_ngw.nat_gateway_id
          - '"state" in first_ngw'
          - first_ngw.state == 'available'
          - '"subnet_id" in first_ngw'
          - first_ngw.subnet_id == subnet_id
          - '"tags" in first_ngw'
          - '"vpc_id" in first_ngw'
          - first_ngw.vpc_id == vpc_id
    vars:
      first_ngw: '{{ avalaible_ngws.result[0] }}'
```

(I'm not a big fan of the lazy 'The result of the describe, converted to ansible snake case style.' results documentation either)

**Review by tremble — CHANGES_REQUESTED** [review]
I'd like to see some form of tests against the shape of the data returned.

Where we search based on something like id, vpc or subnet we should probably check that we've been handed something that matches.

**Review by jillr — COMMENTED** [review]
In addition to the tests you've added I see a few places we can add more assertions to the existing tests to help validate the module's behaviour.

Let's add an assertion to `Search for internet gateway by VPC - no matches` (line 42 task) that igw_info.internet_gateways has a length of zero, to confirm there really are no matches

We should also assert the shape of the return in other places where we create a gateway, like for the task at line 116 (assertion at line 125). That's the first place we create a gw so that's our first chance to check the results that the module returns are accurate.


Anywhere that a NAT gateway is searched for, the assertion should check the ID that was returned is what was expected.

For example at the task at line 150, "Get NAT gateway with specific filters (state and subnet)", the assertion should ensure the correct gateway ID is in the result.
And after line 288, "Fetch NAT gateway by ID (list)".

File: tests/integration/targets/ec2_vpc_nat_gateway/tasks/main.yml (hunk 1 above)
Comment (alinabuzachis) [reply tests/integration/targets/ec2_vpc_nat_gateway/tasks/main.yml:201]: Hi @tremble and @jillr, thank you for reviewing this. I tried to add more assertions even for the other tasks. Let me know please if this makes sense for you or if there is something else I can do to improve the code. Thank you.

Comment (alinabuzachis) [reply tests/integration/targets/ec2_vpc_nat_gateway/tasks/main.yml:201]: @jillr, if the changes I've applied here make sense to you, I'll try to do the same for the internet gateway as well.

Comment (jillr) [reply tests/integration/targets/ec2_vpc_nat_gateway/tasks/main.yml:201]: LGTM - thanks

**Review by jillr — APPROVED** [review]
lgtm, anything you'd like to see @tremble? 
Thanks for this work @alinabuzachis

**Review by tremble — APPROVED** [review]
Looks like a big improvement.  Thanks for your work on this.

**Pull request merged**